src/components/Modals/DeployModal/DeployToWorld_WorldsForEnsOwnersFeature/DeployToWorld.tsx

```diff
@@ -0,0 +1,456 @@
+import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react'
+import classNames from 'classnames'
+import { Button, Field, Icon as DCLIcon, SelectField, Checkbox, Row, Popup, List, DropdownItemProps } from 'decentraland-ui'
+import Modal from 'decentraland-dapps/dist/containers/Modal'
+import { t } from 'decentraland-dapps/dist/modules/translation/utils'
+import { getAnalytics } from 'decentraland-dapps/dist/modules/analytics/utils'
+import { config } from 'config'
+import { isDevelopment } from 'lib/environment'
+import { locations } from 'routing/locations'
+import { Deployment } from 'modules/deployment/types'
+import { FromParam } from 'modules/location/types'
+import CopyToClipboard from 'components/CopyToClipboard/CopyToClipboard'
+import Icon from 'components/Icon'
+import { InfoIcon } from 'components/InfoIcon'
+import { DeployToWorldView, NameType, Props } from './DeployToWorld.types'
+import dclImage from './images/dcl.svg'
+import ensImage from './images/ens.svg'
+
+import styles from './DeployToWorld.module.css'
+
+const EXPLORER_URL = config.get('EXPLORER_URL', '')
+const WORLDS_CONTENT_SERVER_URL = config.get('WORLDS_CONTENT_SERVER', '')
+const CLAIM_NAME_OPTION = 'claim_name_option'
+
+export default function DeployToWorld({
+  name,
+  project,
+  metrics,
+  ensList,
+  externalNames,
+  deployments,
+  isLoading,
+  error,
+  claimedName,
+  onPublish,
+  onRecord,
+  onNavigate,
+  onReplace,
+  onClose,
+  onBack,
+  onFetchExternalNames
+}: Props) {
+  const analytics = getAnalytics()
+
+  const [view, setView] = useState<string>('')
+  const [world, setWorld] = useState<string>(claimedName ?? '')
+  const [nameType, setNameType] = useState<NameType>(NameType.DCL)
+  const [loading, setLoading] = useState<boolean>(false)
+  const [confirmWorldReplaceContent, setConfirmWorldReplaceContent] = useState<boolean>(false)
+  // Ref used to store current world deployment status and validate if the user is trying to deploy the same world
+  const currentDeployment = useRef<Deployment | undefined>()
+
+  const currenWorldLabel = world && ensList.find(ens => ens.subdomain === world)?.name
+
+  useEffect(() => {
+    if (ensList.length === 0) {
+      setView(DeployToWorldView.EMPTY)
+      analytics.track('Publish to World step', { step: DeployToWorldView.EMPTY })
+    } else if (!currentDeployment.current) {
+      setView(DeployToWorldView.FORM)
+      analytics.track('Publish to World step', { step: DeployToWorldView.FORM })
+      onRecord()
+    }
+  }, [ensList, onRecord, analytics])
+
+  useEffect(() => {
+    if (view === DeployToWorldView.FORM && loading && error) {
```

Comment:
Should it show the error view while loading?

Reply:
This logic came from the original component.

What I understand from this is that when the user clicks on publish, loading is set to true.

Then when an error appears in the state, the useEffect picks it and changes the view to ERROR.

On the other case, if no error is provided but the current deployment is found on the store, the SUCCESS view is set.

Reply:
Answering your question: It does not show the error while loading